metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/detection/CriterionTestResult.java

```diff
@@ -148,6 +150,31 @@ public static CriterionTestResult success(final Criterion criterion, final Strin
 			result
 		);
 
-		return CriterionTestResult.builder().success(true).message(message).result(result).build();
+		return CriterionTestResult.builder().success(true).message(message).result(result).criterion(criterion).build();
+	}
+
+	public String displayCriterionMessage() {
+		return String.format(
+			"Executed %s Criterion:\n" +
+			"%s\n" +
+			"\n" +
+			"Result:\n" +
+			"%s\n" +
+			"\n" +
+			"Message:\n" +
+			"====================================\n" +
+			"%s\n" +
+			"====================================\n" +
+			"\n",
+			criterion.getClass().getSimpleName(),
+			criterion.toString(),
+			result != null ? result : "N/A",
+			message != null ? message : "N/A"
+		);
+	}
+
+	public CriterionTestResult setCriterion(final Criterion criterion) {
```

Comment:
Remove this method; we already have a setter. No need to return this here since other setters don’t.